src/ResourceManagement/Compute/WindowsVolumeNoAADEncryptionMonitorImpl.cs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.Compute.Fluent
+{
+    using Microsoft.Azure.Management.Compute.Fluent.Models;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// The implementation for DiskVolumeEncryptionStatus for Windows virtual machine.
+    /// This implementation monitor status of encrypt-decrypt through new NoAAD encryption extension.
+    /// </summary>
+    public partial class WindowsVolumeNoAADEncryptionMonitorImpl :
+        IDiskVolumeEncryptionMonitor
+    {
+        private IComputeManager computeManager;
+        private string rgName;
+        private VirtualMachineInner virtualMachine;
+        private string vmName;
+
+        internal WindowsVolumeNoAADEncryptionMonitorImpl(string virtualMachineId, IComputeManager computeManager)
+        {
+            this.rgName = ResourceUtils.GroupFromResourceId(virtualMachineId);
+            this.vmName = ResourceUtils.NameFromResourceId(virtualMachineId);
+            this.computeManager = computeManager;
+
+        }
+
+        /// <summary>
+        /// Given disk instance view status code, check whether it is encryption status code if yes map it to EncryptionStatus.
+        /// </summary>
+        /// <param name="code">The encryption status code.</param>
+        /// <return>Mapped EncryptionStatus if given code is encryption status code, null otherwise.</return>
+        private static EncryptionStatus EncryptionStatusFromCode(string code)
+        {
+            if (code != null && code.ToLower().StartsWith("encryptionstate"))
```

Comment:
Duplicate code, might factor into the base interface.

Reply:
done